### PR TITLE
Remove generics from application views

### LINKED ISSUE #675: Hide generics from linera-views in wasm applications

### PR DESCRIPTION
**Pull request by jvff**
# Motivation

It can be confusing for new users of the SDK to understand why they need to use generic context types when writing applications with views.

# Solution

Provide view type aliases and alternative derive macros that don't require a generic context type parameter.

# Related Work

Blocked by #727 
Blocked by #737 
Closes #675

### CODE REVIEW

File: linera-views-derive/src/lib.rs [hunk 1]
```diff
@@ -35,16 +38,45 @@ fn get_type_field(field: syn::Field) -> Option<syn::Ident> {
...
+        match specific_context.parse_meta() {
+            Ok(Meta::NameValue(MetaNameValue {
+                lit: Lit::Str(value),
+                ..
+            })) => {
+                context = value.parse().expect("Invalid specific context");
+                constraints = None;
+            }
+            _ => panic!(
+                r#"Invalid `specific_context` attribute syntax. \
```
Comment (ma2bd) [inline linera-views-derive/src/lib.rs:82]: I'm tempted to drop `specific_`.
By analogy with `#[serde(rename = "Bla")]`, shouldn't it be `#[view(context = "Foo")]` ?

File: linera-views-derive/src/lib.rs [hunk 2]
```diff
@@ -35,13 +38,78 @@ fn get_type_field(field: syn::Field) -> Option<syn::Ident> {
...
+            Ok(Meta::NameValue(MetaNameValue {
+                lit: Lit::Str(value),
+                ..
+            })) => {
+                let path: Path = value.parse().expect("Invalid specific context");
+                path.into_token_stream()
+            }
+            _ => panic!(
+                r#"Invalid `crate_path` attribute syntax. \
+                Expected syntax: `#[crate_path = "path::to::linera_views"]`"#,
```
Comment (ma2bd) [inline linera-views-derive/src/lib.rs:55]: Same comment: should it be `#[view(crate_path = "...")]` ?
Also, I couldn't find where this feature is actually used.

**Review by ma2bd — APPROVED** [review]
nice!

File: linera-views-derive/src/lib.rs (hunk 1 above)
Comment (jvff) [reply linera-views-derive/src/lib.rs]: Done.

File: linera-views-derive/src/lib.rs (hunk 2 above)
Comment (jvff) [reply linera-views-derive/src/lib.rs]: Done.

**Pull request merged**